internal/proxy/proxy.go

```diff
@@ -109,3 +136,42 @@ func safeWriteFile(path string, contents string) error {
 	}
 	return nil
 }
+
+// backupExistingFile moves the given file to a backup file suffixed with .old,
+// returning the path to the backup file and a function to restore the original.
+// If the file doesn't exist, no error is returned.
```

Comment:
> // If the file doesn't exist, no error is returned.
I find that behaviour weird though? I would expect the original file not existing at `path` to return an error.